azure/types.go

```diff
@@ -79,6 +79,9 @@ type TagsSpec struct {
 	// The last applied tags are used to find out which tags are being managed by CAPZ
 	// and if any has to be deleted by comparing it with the new desired tags
 	Annotation string
+	// AlwaysManaged, when set to true, ignores the existing tags to determine
+	// whether or not the resource is managed, assuming it is.
+	AlwaysManaged bool
```

Comment:
This is a bit of a hack to handle resources that don't already have the "owned" tag that signals to the tags service that the tags are ok to update. Ideally, this would hook into the resource's service's `IsManaged()` method, but I couldn't think of a good way to do that.

Reply:
How about if we just differentiated based on the presence of a `ManagedClusterTagsLastAppliedAnnotation` annotation in the tagsspec?

Reply:
I agree, still a bit of a hack but I think that would be better contained and harder to misuse down the road.